src/main/java/domain/StringCalcurator.java

```diff
@@ -0,0 +1,93 @@
+package domain;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class StringCalcurator {
+    public List<Integer> getNumbers(List<String> input) {
+        return input.stream()
+                .filter(number -> number.matches("[0-9]"))
+                .map(number -> Integer.parseInt(number))
+                .toList();
+    }
+
+    public List<String> getSigns(List<String> input) {
+        return input.stream()
+                .filter(number -> number.matches("[+\\-/*%]"))
+                .toList();
+    }
+
+    public void validateInput(String input) {
+        if (input == null || input.trim().isEmpty())
+            throw new IllegalArgumentException("입력값이 null이거나 빈 공백 문자입니다.");
+    }
+
+    public int calc(String strInput) {
```

Comment:
`calc` 내부에서 여러 책임을 가지고 있는것 같아요


```suggestion
    public int calc(List<String> val) {
```


```
# 코드1
validateInput(strInput);
List<String> input = Arrays.asList(strInput.split(" "));
validateNumberAndSign(input);
```

이렇게 된다면 `# 코드1` 부분이 검증된 입력을 받게끔 할 수 있을것 같기도 하구요,
위 코드는 예시일뿐입니다! (정답은 없으니 고민해주세요~)

[[클린 코드: Clean Code] 3. 함수(Function)](https://data-make.tistory.com/633) 참고해보시면 좋을것같아요!

Reply:
```suggestion
        private List<String> getListString(String input) {
        return Arrays.asList(input.split(" "));
    }

    public int calc(String strInput) {
        validateInput(strInput);
        List<String> input = getListString(strInput);
        validateNumberAndSign(input);

        return calc(input);
    }

    private int calc(List<String> input) {
        List<Integer> numbers = getNumbers(input);
        List<String> signs = getSigns(input);

        return initCalcurator(numbers, signs);
    }
```

해당 내용 참고하여 수정했습니다.
StringCalcurator는 리뷰 대상으로 포함되지 않는다고 들어서 간단히 테스트용으로 작성된 소스인점 참고 부탁드립니다.